### ISSUE TITLE
Some rules tried to validate non react files

### ISSUE THREAD

**Issue by MoOx** [yannickcr/eslint-plugin-react#262]
I added some new rules, then got a lot of issues like this

```
Users/MoOx/Sync/Development/statinamic/src/configurator/__tests__/index.js
  1:1  error  Definition for rule 'react/no-direct-mutation-state' was not found  react/no-direct-mutation-state
  1:1  error  Definition for rule 'prefer-es6-class' was not found                prefer-es6-class
```

Those 2 rules `react/no-direct-mutation-state` and `prefer-es6-class` are trying to validate non react files. And for tests files for example, it's not really relevant. Do you think that can be improved?

**Comment by MoOx** [comment]
Oups. I will edit the description then delete this comment.

**Comment by MoOx** [comment]
FYI, this file looks like this

```js
import tape from "tape"

import configurator from ".."

tape("statinamic/lib/configurator", (test) => {

  const pkg = {
    homepage: "https://localhost:8080/statinamic",
  }
  const config = configurator(pkg)
  test.equal(
    typeof config.baseUrl,
    "object",
    "should return an 'baseUrl' object in the config"
  )
  test.equal(
    config.baseUrl.hostname,
    "localhost",
    "should contain 'hostname' in 'baseUrl'"
  )
  test.equal(
    config.baseUrl.port,
    "8080",
    "should contain 'port' in 'baseUrl'"
  )
  test.equal(
    config.baseUrl.pathname,
    "/statinamic",
    "should contain 'pathname' in 'baseUrl'"
  )

  test.equal(
    typeof config.consts,
    "object",
    "should return an 'consts' object in the config"
  )

  test.end()
})
```

**Comment by yannickcr** [comment]
Even if theses are non react files, you should not get some errors.

The message `Definition for rule 'xxx' was not found` suggest that ESLint cannot find theses rules in its default rules or in the plugin.

* `react/no-direct-mutation-state` is only available since eslint-plugin-react `3.5.0`
* `prefer-es6-class` misses the react prefix, it should be `react/prefer-es6-class`. It is available since eslint-plugin-react `3.6.0`

Also, if you want to deactivate some rules in your test directory you can use the [configuration cascading](http://eslint.org/docs/user-guide/configuring#configuration-cascading-and-hierarchy).

**Comment by MoOx** [comment]
Oh sorry, I must have not updated correctly my version.
